samza-core/src/main/java/org/apache/samza/operators/stream/IntermediateMessageStreamImpl.java

```diff
@@ -39,12 +41,19 @@
  */
 public class IntermediateMessageStreamImpl<M> extends MessageStreamImpl<M> implements OutputStream<M> {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(IntermediateMessageStreamImpl.class);
   private final OutputStreamImpl<M> outputStream;
+  private final boolean isKeyed;
 
   public IntermediateMessageStreamImpl(StreamGraphImpl graph, InputOperatorSpec<?, M> inputOperatorSpec,
       OutputStreamImpl<M> outputStream) {
     super(graph, (OperatorSpec<?, M>) inputOperatorSpec);
     this.outputStream = outputStream;
+    if (inputOperatorSpec.isKeyed() != outputStream.isKeyed()) {
+      LOGGER.error("Input and output streams for intermediate stream {} aren't keyed consistently. Input: {}, Output: {}",
```

Comment:
Is this error actionable by the user? they should also know that there is an input stream, and output stream corresponding to every intermediate stream?

Reply:
No, this would be a framework error. Just a defensive check for us to help debug in case this happens.